cou/steps/ceph.py

```diff
@@ -0,0 +1,270 @@
+        )
+    logger.debug("Currently OSDs are on the '%s' release", current_osd_release)
+
+    return current_osd_release
+
+
+async def _get_applications(model: Model) -> list[Application]:
+    """Get the all ceph mon application(s) from the model.
+
+    :param model: The juju model to work with
+    :type model: Model
+    :return: A list of ceph-mon applications
+    :type: Application
+    :raise ApplicationNotFound: if ceph-mon no founnd
+    """
+    apps = await model.get_applications()
+    ceph_mon_apps = [app for app in apps.values() if app.charm == "ceph-mon"]
+    if not ceph_mon_apps:
+        raise ApplicationNotFound(
+            "'ceph-mon' application not found, is this a valid OpenStack cloud?"
+        )
+    return ceph_mon_apps
+
+
+async def _get_unit_name(ceph_mon_app: Application) -> str:
+    """Get the one of the unit's name from ceph mon application.
+
+    :param ceph_mon_app: The ceph mon application
+    :type ceph_mon_app: Application
+    :return: ceph-mon's unit name
+    :type: str
+    :raise UnitNotFound: if ceph-mon has no units
+    """
+    ceph_mon_units = list(ceph_mon_app.units.values())
+    if not ceph_mon_units:
+        raise UnitNotFound(f"'{ceph_mon_app.name}' has no units, is this a valid OpenStack cloud?")
+    return ceph_mon_units[0].name
+
+
+async def osd_noout(model: Model, enable: bool) -> None:
+    """Set or unset 'noout' for ceph cluster(s).
+
+    :param model: The juju model to work with
+    :type model: Model
+    :param enable: True to set noout, False to unset noout
+    :type enable: bool
+    """
+    try:
+        ceph_mon_apps = await _get_applications(model)
+    except ApplicationNotFound as e:
+        logger.warning("%s", str(e))
+        logger.warning("Skip changing 'noout', because there's no ceph-mon applications.")
+        return
+
+    for ceph_mon_app in ceph_mon_apps:
+        try:
+            ceph_mon_unit_name = await _get_unit_name(ceph_mon_app)
+        except UnitNotFound as e:
+            logger.warning("%s", str(e))
+            logger.warning(
+                "Skip changing 'noout', because there's no %s units.", ceph_mon_app.name
+            )
+            continue
+
+        await model.run_action(
+            ceph_mon_unit_name, f"{'set' if enable else 'unset'}-noout", raise_on_failure=True
+        )
+
+
+async def get_osd_noout_state(model: Model, unit_name: str) -> bool:
+    """Check if noout is set on ceph cluster or not.
+
+    :param model: The juju model to work with
+    :type model: Model
+    :param unit_name: The name of the unit
+    :type unit_name: str
+    :return: True if noout is set, otherwise False
+    :type: bool
+    """
+    results = await model.run_on_unit(unit_name, "ceph osd dump -f json")
+    return "noout" in json.loads(results["stdout"].strip()).get("flags_set", [])
+
+
+async def assert_osd_noout_state(model: Model, state: bool) -> None:
+    """Assert ceph cluster is set (state=True) or unset (state=False).
+
```

Comment:
Should mention in the docstring that this works on all ceph-mon apps - same for the other functions that iterate over all the ceph-mon apps.